frame/contracts/src/wasm/code_cache.rs

```diff
@@ -95,8 +96,26 @@ pub fn store<T: Config>(mut module: PrefabWasmModule<T>, instantiated: bool) ->
 				);
 				// This `None` case happens only in freshly uploaded modules. This means that
 				// the `owner` is always the origin of the current transaction.
-				T::Currency::reserve(&new_owner_info.owner, new_owner_info.deposit)
-					.map_err(|_| <Error<T>>::StorageDepositNotEnoughFunds)?;
+				T::Currency::hold(
+					&T::HoldReason::get(),
+					&new_owner_info.owner,
+					new_owner_info.deposit,
+				)
+				.map_err(|e| match e {
+					DispatchError::Token(TokenError::FundsUnavailable) =>
+						<Error<T>>::StorageDepositNotEnoughFunds.into(),
+					_ => e,
+				})?;
+				<Pallet<T>>::deposit_event(
+					vec![
+						T::Hashing::hash_of(&new_owner_info.owner),
+						T::Hashing::hash_of(&new_owner_info.deposit),
+					],
+					Event::StorageDepositHeld {
+						who: new_owner_info.owner.clone(),
+						amount: new_owner_info.deposit.clone(),
+					},
```

Comment:
The old `Event::Reserved` had an empty vec as a `topic`, not sure if we want to keep the old behaviour or it makes sense to add some topics. Not sure if it even makes sense to have the deposit as a topic.
Happy to hear some thoughts on this

Reply:
This isn't longer valid as the file has been removed https://github.com/paritytech/substrate/pull/14084